File: leetCode/121_maxProfit.py
class Solution:
   def maxProfit(self, prices: List[int]) -> int:
      max_profit = 0
      curr_lowest_price = prices[0]

      for i in range(1, len(prices)):
         if prices[i] < curr_lowest_price:
            curr_lowest_price = prices[i]
         else:
            max_profit = max(max_profit, prices[i] - curr_lowest_price)

      return max_profit

File: leetCode/121_maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
   if (prices.length < 2) return 0;

   let left = 0;
   let right = 1;
   let maxProfit = 0;

   for (let i = 1; i < prices.length; i++) {
      if (prices[right] > prices[left]) {
         let profit = prices[right] - prices[left];
         maxProfit = Math.max(profit, maxProfit);
      } else {
         left = right;
      }
      right++;
   }

   return maxProfit
};
